### PR TITLE
feat(Trend Micro/Cloud One): Init Cloud One format

### PR DESCRIPTION
**Pull request by PierrickV**
Parent task:
- https://github.com/SekoiaLab/platform/issues/30935

### CODE REVIEW

File: Trend Micro/cloud-one/ingest/parser.yml [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+  set_cloud_one_custom_fields:
+    actions:
+      - set:
+          trendmicro: '{{parsed_event.message}}'
+          trendmicro.LogInspectionDescription: '{{parsed_event.message.get("LI Description")}}'
+          trendmicro.IntrusionPreventionStreamPosition: '{{parsed_event.message.get("Intrusion Prevention Stream Position")}}'
+          trendmicro.IntrusionPreventionFlags: '{{parsed_event.message.get("Intrusion Prevention Flags")}}'
+          trendmicro.FragmentationBits: '{{parsed_event.message.get("Fragmentation Bits")}}'
+          trendmicro.TCPFlags: '{{parsed_event.message.get("TCP Flags")}}'
+          container.name:  '{{parsed_event.message.get("Container)}}'
```
Comment (squioc) [inline Trend Micro/cloud-one/ingest/parser.yml:37]:
```suggestion
          container.name:  '{{parsed_event.message.get("Container")}}'
```

File: Trend Micro/cloud-one/ingest/parser.yml [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+  - name: set_cef_header_fields
+  - name: set_cef_extension_fields
+  - name: set_cloud_one_custom_fields
+
+stages:
+  set_ecs_categorization_fields:
+    actions:
+      - set:
+          event.kind: 'event'
+          event.category: 'intrusion_detection'
```
Comment (squioc) [inline Trend Micro/cloud-one/ingest/parser.yml:17]:
```suggestion
          event.category: ['intrusion_detection']
```

File: Trend Micro/cloud-one/tests/SystemEventLog.json [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    "sekoiaio": {
+      "intake": {
+        "dialect": "Cloud One",
+        "dialect_uuid": "b2d961ae-0f7e-400b-879a-f97be24cc02d"
+      }
+    },
+    "message": "CEF:0|Trend Micro|Workload Security Manager|<Workload Security version>|600|User Signed In|3|src=10.52.116.160 suser=admin target=admin msg=User signed in from 2001:db8::5"
+  },
+  "expected": {
+    "message": "CEF:0|Trend Micro|Workload Security Manager|<Workload Security version>|600|User Signed In|3|src=10.52.116.160 suser=admin target=admin msg=User signed in from 2001:db8::5",
```
Comment (squioc) [inline Trend Micro/cloud-one/tests/SystemEventLog.json:12]: Not an intrusion detection but a sign-in.
The category of the event should be ['authentication'] and the type ['start']

File: Trend Micro/cloud-one/tests/IntrusionPrevention.json [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+      "category": "intrusion_detection",
+      "severity": 3,
+      "action": "IDS:Reset"
+    },
+    "trendmicro": {
+      "TrendMicroDsFrameType": "IP",
+      "IntrusionPreventionStreamPosition": "10",
+      "IntrusionPreventionFlags": "8",
+      "FragmentationBits": "DF MF",
+      "TCPFlags": "0x00 ACK PSH",
```
Comment (squioc) [inline Trend Micro/cloud-one/tests/IntrusionPrevention.json:24]: I think this field could be an array

File: Trend Micro/cloud-one/tests/AgentAntiMalware.json [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    "message": "CEF:0|Trend Micro|Deep Security Agent|<Agent version>|4000000|Eicar_test_file|6|cn1=1 cn1Label=Host ID dvchost=hostname cn2=205 cn2Label=Quarantine File Size cs6=ContainerImageName | ContainerName | ContainerID cs6Label=Container filePath=C:\\Users\\trend\\Desktop\\eicar.exe act=Delete result=Delete msg=Realtime TrendMicroDsMalwareTarget=N/A TrendMicroDsMalwareTargetType=N/A TrendMicroDsFileMD5=44D88612FEA8A8F36DE82E1278ABB02F TrendMicroDsFileSHA1=3395856CE81F2B7382DEE72602F798B642F14140 TrendMicroDsFileSHA256=275A021BBFB6489E54D471899F7DB9D1663FC695EC2FE2A2C4538AABF651FD0F TrendMicroDsDetectionConfidence=95 TrendMicroDsRelevantDetectionNames=Ransom_CERBER.BZC;Ransom_CERBER.C;Ransom_CRYPNISCA.SM",
+    "event": {
+      "kind": "event",
+      "category": "intrusion_detection",
+      "severity": 6,
+      "action": "Delete",
+      "reason": "Realtime"
+    },
+    "trendmicro": {
+      "TrendMicroDsRelevantDetectionNames": "Ransom_CERBER.BZC;Ransom_CERBER.C;Ransom_CRYPNISCA.SM",
```
Comment (squioc) [inline Trend Micro/cloud-one/tests/AgentAntiMalware.json:21]: I think this field would be more suitable as an array

**Review by squioc — CHANGES_REQUESTED** [review]

File: Trend Micro/cloud-one/tests/SystemEventLog.json (hunk 3 above)
Comment (PierrickV) [reply Trend Micro/cloud-one/tests/SystemEventLog.json:12]: Fixed but we will need more log to be more precise

File: Trend Micro/README.md [hunk 6]
```diff
@@ -0,0 +1,7 @@
+# Trend Micro
+
+## Description
+Trend Micro develops enterprise security software for servers, containers, & cloud computing environments, networks, and end points. Its cloud and virtualization security products provide automated security for customers of VMware, Amazon AWS, Microsoft Azure, and Google Cloud Platform ([Wikipedia](https://en.wikipedia.org/wiki/Trend_Micro)).
```
Comment (squioc) [inline Trend Micro/README.md:4]:
```suggestion
Trend Micro develops enterprise security software for servers, containers, & cloud computing environments, networks, and end points. Its cloud and virtualization security products provide automated security for customers of VMware, Amazon AWS, Microsoft Azure, and Google Cloud Platform.
```

File: Trend Micro/_meta/manifest.yml [hunk 7]
```diff
@@ -0,0 +1,5 @@
+uuid: afc0e987-ba67-4979-a31f-012c4657f523
+name: Trend Micro
+slug: trend-micro
+description: >-
+  Trend Micro develops enterprise security software for servers, containers, & cloud computing environments, networks, and end points. Its cloud and virtualization security products provide automated security for customers of VMware, Amazon AWS, Microsoft Azure, and Google Cloud Platform ([Wikipedia](https://en.wikipedia.org/wiki/Trend_Micro)).
```
Comment (squioc) [inline Trend Micro/_meta/manifest.yml:5]:
```suggestion
  Trend Micro develops enterprise security software for servers, containers, & cloud computing environments, networks, and end points. Its cloud and virtualization security products provide automated security for customers of VMware, Amazon AWS, Microsoft Azure, and Google Cloud Platform.
```

**Review by squioc — APPROVED** [review]

**Review by squioc — APPROVED** [review]

**Pull request merged**